app/models/ncr_dispatcher.rb

```diff
@@ -1,67 +0,0 @@
-class NcrDispatcher < LinearDispatcher
```

Comment:
I agree that this PR clarifies a lot and on the whole I like it.

I'm wondering if we could just create a `NcrDispatcher` subclass that only overrides the methods where you have `ncr?` checks. Then it would be up to the calling code to call the client-specific `NcrDispatcher` rather than `Distpatcher`.